specs/experimental/contracts/L1/eth-lockbox.md

```diff
@@ -0,0 +1,257 @@
+- [Assumptions](#assumptions)
+  - [a01-001: ProxyAdmin Owner Governance](#a01-001-proxyadmin-owner-governance)
+    - [Mitigations](#mitigations)
+- [Invariants](#invariants)
+  - [i01-001: Unified Liquidity Sufficiency](#i01-001-unified-liquidity-sufficiency)
+    - [Impact](#impact)
+  - [i01-002: Withdrawal Transaction Isolation](#i01-002-withdrawal-transaction-isolation)
+    - [Impact](#impact-1)
+  - [i01-003: Complete Liquidity Migration](#i01-003-complete-liquidity-migration)
+    - [Impact](#impact-2)
+- [Function Specification](#function-specification)
+  - [initialize](#initialize)
+  - [lockETH](#locketh)
+  - [unlockETH](#unlocketh)
+  - [authorizePortal](#authorizeportal)
+  - [authorizeLockbox](#authorizelockbox)
+  - [migrateLiquidity](#migrateliquidity)
+  - [receiveLiquidity](#receiveliquidity)
+  - [paused](#paused)
+  - [superchainConfig](#superchainconfig)
+  - [proxyAdminOwner](#proxyadminowner)
+  - [proxyAdmin](#proxyadmin)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Manages unified ETH liquidity for multiple OptimismPortal contracts within a Superchain cluster,
+enabling withdrawals from any chain regardless of where ETH was originally deposited.
+
+## Definitions
+
+### Authorized Portal
+
+An OptimismPortal contract that has been granted permission to lock and unlock ETH from the
+lockbox. Authorization requires the portal to share the same ProxyAdmin owner and SuperchainConfig
+as the lockbox.
+
+### Authorized Lockbox
+
+Another ETHLockbox contract that has been granted permission to migrate its liquidity to the
+current lockbox. Authorization requires both lockboxes to share the same ProxyAdmin owner.
+
+### Liquidity Migration
+
+The process of transferring the entire ETH balance from one ETHLockbox to another, typically
+performed during system upgrades or when merging multiple lockboxes into a unified liquidity pool.
+
+## Assumptions
+
+### a01-001: ProxyAdmin Owner Governance
+
+The ProxyAdmin owner operates within governance constraints when authorizing portals and lockboxes,
+and when initiating liquidity migrations.
+
+#### Mitigations
+
+- Authorization functions are restricted to ProxyAdmin owner only
+- Shared ProxyAdmin owner requirement ensures consistent security model across components
+- SuperchainConfig consistency check prevents cross-cluster authorization
+
+## Invariants
+
+### i01-001: Unified Liquidity Sufficiency
+
+The ETH balance held by the lockbox must always be sufficient to fulfill all pending withdrawal
+obligations across all authorized portals. The total withdrawable ETH across all chains in the
+dependency set must never exceed the lockbox balance.
+
+#### Impact
+
+**Severity: Critical**
+
+If violated, legitimate withdrawals could fail due to insufficient liquidity, breaking the core
+promise of unified liquidity management and potentially trapping user funds.
+
+### i01-002: Withdrawal Transaction Isolation
```

Comment:
devinai yet another implementation detail as an invariant. plz fix the prompt to avoid doing this in the future.

plz:

assumptions:
- owner is governance trusted
- approved Portal contracts are trusted and will only request valid withdrawals of funds

invariants:
- only distributes to approved portals

that simple